rooms.js

```diff
@@ -933,16 +933,34 @@ class BattleRoom extends Room {
 	}
 	requestKickInactive(user, force) {
 		if (this.resetTimer) {
-			if (user) this.sendUser(user, '|inactive|The inactivity timer is already counting down.');
-			return false;
+			let timerSetByUser = user && (this.resetUser === user.userid || (user in this.game.players && this.resetUser === '+'));
```

Comment:
@MacChaeger - Zarel is saying that he'd probably want to do the refactor first, and then have you rebase this with the master branch _after_ his refactor so that it doesn't further complicate the rewrite. 

When he does, it will say on here that there are conflicts that need to be resolved, and at that point, you can then rebase it, and if you need help, then ask me. :)